### PR TITLE
[master < T0712-MG] Add support for temporal types in the communication layer of mgclient

### PR DESCRIPTION
**Pull request by kostasrim**
- [ x ] Complete the communication layer to support TT

### CODE REVIEW

File: src/mgvalue.c [hunk 1]
```diff
@@ -1639,6 +1640,48 @@ mg_path *mg_path_make(uint32_t node_count, mg_node **nodes,
...
+mg_duration *mg_duration_make(int64_t months, int64_t days, int64_t seconds,
+                              int64_t nanoseconds) {
+  mg_duration *dur = mg_duration_alloc(&mg_system_allocator);
+  if (!dur) {
+    return NULL;
+  }
+  dur->months = months;
+  dur->days = days;
+  dur->seconds = seconds;
+  dur->nanoseconds = nanoseconds;
```
Comment (jbajic) [inline src/mgvalue.c:1681]: Aren't we also using microseconds, or are those encompassed in the nanoseconds?

**Review by antaljanosbenjamin — APPROVED** [review]

File: src/mgvalue.c (hunk 1 above)
Comment (kostasrim) [reply src/mgvalue.c:1681]: Good question! Anything below seconds is converted into nanoseconds and sent over the wire. The caller of this function must make this conversion.

**Review by jbajic — APPROVED** [review]

File: src/mgvalue.c (hunk 1 above)
Comment (antonio2368) [reply src/mgvalue.c:1681]: everything here is the implementation of the PackStream for temporal types:
https://7687.org/packstream/packstream-specification-1.html#localdatetime---structure

**Review by antonio2368 — APPROVED** [review]

File: include/mgclient.h [hunk 2]
```diff
@@ -947,6 +955,11 @@ MGCLIENT_EXPORT mg_date_time_zone_id *mg_date_time_zone_id_copy(
 MGCLIENT_EXPORT void mg_date_time_zone_id_destroy(
     mg_date_time_zone_id *date_time_zone_id);
 
+/// Creates mg_local_date_time from seconds and nanoseconds.
+/// \return a pointer to mg_local_date_time or NULL if an error occurred.
+MGCLIENT_EXPORT mg_local_date_time *mg_local_date_time_make(
```
[the remark this answers is not in the document]
Comment (kostasrim) [reply include/mgclient.h:960]: Yes, it was not correct. I missed that, I edited it :muscle:

**Pull request merged**